### PR TITLE
Add COVAR_POP, COVAR_SAMP, CORR, REGR_SLOPE, REGR_INTERCEPT

### PR DESCRIPTION
**Pull request by haozhun**
Supersedes #2051

### CODE REVIEW

**Comment by haozhun** [comment]
@electrum This pull request should have fixed all comments you pointed out in #2051. Otherwise, it is the same as #2051 except for bring everything up to date.

**Comment by electrum** [comment]
`covar_pop` seems to be missing

**Comment by electrum** [comment]
Add release notes

File: presto-docs/src/main/sphinx/functions/aggregate.rst [hunk 1]
```diff
@@ -139,6 +139,22 @@ Approximate Aggregate Functions
 Statistical Aggregate Functions
 -------------------------------
 
+.. function:: corr(x, y) -> double
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/aggregate.rst]: This is missing `covar_pop`

Comment (haozhun) [inline presto-docs/src/main/sphinx/functions/aggregate.rst]: It's there now

**Comment by haozhun** [comment]
* Fixed mathematical bug in original code
* Made tests non-trivial
* Added missing covar_pop
* Made edge case handling agree to standard
* Simplified math

**Comment by haozhun** [comment]
Ready for review

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/CorrelationAggregation.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+@AggregationFunction("corr")
+public class CorrelationAggregation
+{
+    private CorrelationAggregation() {}
+
+    @InputFunction
+    public static void input(CorrelationState state, @SqlType(StandardTypes.DOUBLE) double independentValue, @SqlType(StandardTypes.DOUBLE) double dependentValue)
+    {
+        updateCorrelationState(state, independentValue, dependentValue);
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/CorrelationAggregation.java]: name these `x` and `y`

Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/CorrelationAggregation.java]: nevermind, the spec uses those terms

Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/CorrelationAggregation.java]: OK. That's where independent and dependent come from.

Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/CorrelationAggregation.java]: Swap the names. The standard refers to the first value as the dependent value

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/CovarianceAggregation.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import static com.facebook.presto.operator.aggregation.AggregationUtils.updateCovarianceState;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+
+@AggregationFunction("")
+public class CovarianceAggregation
+{
+    private CovarianceAggregation() {}
+
+    @InputFunction
+    public static void input(CovarianceState state, @SqlType(StandardTypes.DOUBLE) double independentValue, @SqlType(StandardTypes.DOUBLE) double dependentValue)
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/CovarianceAggregation.java]: Swap the order of the arguments

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/RegressionAggregation.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+@AggregationFunction("") // Names are on output methods
+public class RegressionAggregation
+{
+    private RegressionAggregation() {}
+
+    @InputFunction
+    public static void input(RegressionState state, @SqlType(StandardTypes.DOUBLE) double independentValue, @SqlType(StandardTypes.DOUBLE) double dependentValue)
+    {
+        updateRegressionState(state, independentValue, dependentValue);
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/RegressionAggregation.java]: This is incorrect. The first argument is the dependent value

File: presto-docs/src/main/sphinx/functions/aggregate.rst (hunk 1 above)
Comment (cberner) [inline presto-docs/src/main/sphinx/functions/aggregate.rst]: All these arguments are in the wrong order. `y` should be the first argument. See `binary set function` in the grammer of the SQL spec

File: presto-docs/src/main/sphinx/functions/aggregate.rst [hunk 5]
```diff
@@ -139,6 +139,26 @@ Approximate Aggregate Functions
...
+
+    Returns the population covariance of input values.
+
+.. function:: covar_samp(x, y) -> double
+
+    Returns the sample covariance of input values.
+
+.. function:: regr_intercept(x, y) -> double
+
+    Returns linear regression intercept of input values.
```
Comment (cberner) [inline presto-docs/src/main/sphinx/functions/aggregate.rst]: clarify that y is the dependent variable and swap the order of the arguments

File: presto-docs/src/main/sphinx/functions/aggregate.rst [hunk 6]
```diff
@@ -139,6 +139,26 @@ Approximate Aggregate Functions
...
+
+.. function:: regr_slope(x, y) -> double
+
+    Returns linear regression slope of input values.
```
Comment (cberner) [inline presto-docs/src/main/sphinx/functions/aggregate.rst]: clarify that y is the dependent variable and swap the order of the arguments

**Comment by cberner** [comment]
Assign back to me, after you've fixed the argument order

**Comment by cberner** [comment]
Couple minor comments. Looks good
